pandas/tests/frame/methods/test_replace.py

```diff
@@ -1420,3 +1478,31 @@ def test_replace_period_ignore_float(self):
         result = df.replace(1.0, 0.0)
         expected = pd.DataFrame({"Per": [pd.Period("2020-01")] * 3})
         tm.assert_frame_equal(expected, result)
+
+    def test_replace_value_category_type(self, input_category_df, expected_category_df):
+        """
+        Test to ensure category dtypes are maintained
+        after replace with direct values
+        """
+
+        # replace values in input dataframe
+        input_df = input_category_df.replace("d", "z")
+        input_df = input_df.replace("obj1", "obj9")
+        input_df = input_df.replace("cat2", "catX")
+
+        tm.assert_frame_equal(input_df, expected_category_df)
+
+    @pytest.mark.xfail(
+        reason="category dtype gets changed to object type after replace, see #23305",
+        strict=True,
+    )
+    def test_replace_dict_category_type(self, input_category_df, expected_category_df):
+        """
+        Test to ensure category dtypes are maintained
```

Comment:
just make a simple example that you can inline here, no need for the fixtures. Please create a new issue for this case (as we will close the original with this PR). indicate the issue number in the xfail

Reply:
I have raised a new issue for the bug - https://github.com/pandas-dev/pandas/issues/35268 and referenced it in the xfail.

Reply:
Hi @jreback, please let me know if we need anything else to merge this PR. Thanks!